jetstream_pt/layers.py

```diff
@@ -367,9 +367,25 @@ def apply_rotary_emb(
 def repeat_kv(x: torch.Tensor, n_rep: int) -> torch.Tensor:
   """torch.repeat_interleave(x, dim=2, repeats=n_rep)."""
 
-  bs, n_kv_heads, slen, head_dim = x.shape
+  bs, n_kv_heads, slen, head_dim = (
+      x.shape[-4],
+      x.shape[-3],
+      x.shape[-2],
+      x.shape[-1],
+  )
+  if x.ndim == 5:
+    stacked = True
+  else:
+    stacked = False
   if n_rep == 1:
     return x
+  if stacked:
```

Comment:
or just put the ndim == 5 here and remove the stacked var

Reply:
I'd probably prefer to keep stacked to make the code more clear.